### PR TITLE
fix: adjust quorum rotation data results in some edge cases, add tests

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
Before this PR you could get `baseBlockHash` as all 0s for some cases. Also, a known `baseBlockHash` that matched the `cycleHash` was ignored which means that results included quorum changes from an older known `baseBlockHash` till the `cycleHash`(which is the same as the `baseBlockHash` in question) i.e. these changes were already known to the requester.

## What was done?
Pls see individual commits

## How Has This Been Tested?
Run tests

## Breaking Changes
n/a, results should simply be a bit more meaningful and compact

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: test/functional/feature_llmq_rotation.py [hunk 1]
```diff
@@ -231,6 +231,23 @@ def run_test(self):
         self.wait_until(lambda: self.nodes[0].getbestblockhash() == new_quorum_blockhash)
         assert_equal(self.nodes[0].quorum("list", llmq_type), new_quorum_list)
 
+        self.log.info("Test 'quorum rotationinfo' RPC")
+        genesis_blockhash = self.nodes[0].getblockhash(0)
+        block_count = self.nodes[0].getblockcount()
+        hmc_base_blockhash = self.nodes[0].getblockhash(block_count - (block_count % 24) - 24 - 8)
+        best_block_hash = self.nodes[0].getbestblockhash()
+        rpc_qr_info = self.nodes[0].quorum("rotationinfo", best_block_hash, False, hmc_base_blockhash)
+        self.log.info(rpc_qr_info)
```
Comment (knst) [inline test/functional/feature_llmq_rotation.py:240]: I think there's no use to log it as it is; it looks like a leftover  from writing & debugging.

```diff
-self.log.info(rpc_qr_info)
```

File: src/llmq/snapshot.cpp [hunk 2]
```diff
@@ -392,13 +392,14 @@ bool BuildQuorumRotationInfo(CDeterministicMNManager& dmnman, CQuorumSnapshotMan
 uint256 GetLastBaseBlockHash(Span<const CBlockIndex*> baseBlockIndexes, const CBlockIndex* blockIndex,
                              bool use_legacy_construction)
 {
-    uint256 hash;
     if (!use_legacy_construction) {
         std::sort(baseBlockIndexes.begin(), baseBlockIndexes.end(),
                   [](const CBlockIndex* a, const CBlockIndex* b) { return a->nHeight < b->nHeight; });
     }
+    // default to genesis block
+    uint256 hash{Params().GenesisBlock().GetHash()};
```
[the remark this answers is not in the document]
Comment (knst) [reply src/llmq/snapshot.cpp:400]: why do we need to set genesis block here? Is it still possible situation when hash can be _default_?

**Review by knst — APPROVED** [review]
LGTM 8b9a728de4dae0971029a9376a77f13c81aa5bbd

**Review by PastaPastaPasta — APPROVED** [review]
utACK 8b9a728de4dae0971029a9376a77f13c81aa5bbd
